src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs

```diff
@@ -355,14 +356,38 @@ public static void AddTopLevelSuggestionsForCursorType(IntellisenseData.Intellis
 
                     if (!intellisenseData.TryAddSuggestionForCurrentBinaryOp(checkForOptionSetOnly: true))
                     {
-                        if (info.ScopeIdentifier != default)
+                        var node = callNode.Args.Children[argPosition];
+
+                        // filter on what's already typed (up to the cursor position)
+                        int start = node.Token.Span.Min;
+                        int end = node.Token.Span.Lim;
+
+                        // if node is BinaryOpNode, use the Right part for determining the filter
+                        if (node is BinaryOpNode binOpNode)
                         {
+                            start = binOpNode.Right.Token.Span.Min;
+                            end = binOpNode.Right.Token.Span.Lim;
+                        }
+
```

Comment:
If there is specific logic to binaryopnode can we possibly incorporate this in TryAddSuggestionForCurrentBinaryOp?

Reply:
Removed